src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -247,6 +247,23 @@ static int fstatat_wrapper(int dfd, const char *path,
 }
 #endif
 
+#if defined(__linux__)
+/**
+ * Lookup functions with time_t parameter. Try to use 64 bit symbol
+ * if sizeof(time_t) exceeds 32 bit.
+ */
+static void* lookup_time_t_function(const char* symbol, const char* symbol64) {
+    void* ret = NULL;
+    if (sizeof(time_t) > 4) {
+        ret = dlsym(RTLD_DEFAULT, symbol64);
+    }
+    if (ret == NULL) {
+        return dlsym(RTLD_DEFAULT, symbol);
+    }
+    return ret;
+}
+#endif
```

Comment:
I assume you will move to after statx_wrapper and just replace the body with a simple if-then-else.

Reply:
if/else will not work in a general case, even in Debian where x32 arch has 64 bit time_t but no 64 bit functions, so I have to lookup 64 bit one and, if it fails, fall back to a general symbol.

Reply:
Starting out with this limited to ARM32 is okay, it reduces the risk as this code is compiled to many platforms and with many different toolchains. Can you move it after statx_wrapper as right now it is a bit annoying to have in the middle of the XXX_wrapper functions.